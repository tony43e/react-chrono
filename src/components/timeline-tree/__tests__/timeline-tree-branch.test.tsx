import { render, screen, fireEvent } from "@testing-library/react";
import React from "react";
import Branch from "../timeline-tree-branch";

const onClick = jest.fn();
const onActive = jest.fn();
const showMore = jest.fn();

const View = (
  <Branch
    className="test-class"
    index={2}
    contentText="Tree branch test"
    title="branch title"
    id="223344"
    active
    onClick={() => onClick("zz22ww")}
    onActive={onActive}
    onShowMore={showMore}
  />
);

test("Test tree branch render", () => {
  render(View);
  const element = screen.getByTestId("branch-main");
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});

test("Test tree branch class styles", () => {
  render(View);
  const element = screen.getByTestId("branch-main");
  expect(element).toHaveClass("test-class");

  Array.from(element.children).forEach((child, index) =>  {
    expect(child).toHaveClass("test-class");

    if(index < 2) {
      expect(child.children[0]).toHaveClass("active");
    }
  })
});

test("Test tree branch render", () => {
  render(View);
  const element = screen.getByTestId("tree-leaf-click");

  expect(element).toBeInTheDocument();
  fireEvent.click(element);

  expect(onClick).toBeCalled();
  expect(onActive).toBeCalled();
  expect(onClick).toBeCalledWith("zz22ww");
  
});
