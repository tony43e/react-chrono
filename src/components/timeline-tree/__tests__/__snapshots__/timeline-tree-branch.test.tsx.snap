// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test tree branch render 1`] = `
<div
  class="test-class  branch-main css-1uu8hqk"
  data-testid="branch-main"
>
  <div
    class="test-class css-1sgx6fg"
  >
    <div
      class="active css-x77ob3"
      title="branch title"
    >
      branch title
    </div>
  </div>
  <div
    class="test-class  css-1yrljia"
  >
    <div
      class="active css-74e5rn"
    >
      <span
        class="css-c1e5f8"
      >
        Tree branch test
      </span>
      <div
        class="show-less css-or8e72"
      />
      <span
        class="show-more css-1rml49d"
        role="button"
      >
        show more
      </span>
    </div>
  </div>
  <div
    class="test-class css-ey985k"
    data-testid="tree-leaf"
  >
    <div
      class="test-class css-kooiip"
      data-testid="tree-leaf-click"
    >
      <div
        class="active css-1yzc5y5"
      />
    </div>
  </div>
</div>
`;
